test/js_test.go

```diff
@@ -1346,51 +1346,166 @@ func TestJetStreamManagement(t *testing.T) {
+				t.Fatalf("Unexpected error: %v", err)
+			}
+			msg, err := sub.NextMsg(1 * time.Second)
+			if err != nil {
+				t.Fatalf("Unexpected error: %v", err)
+			}
+			if !strings.Contains(string(msg.Data), `"durable_name":"dlc-1"`) {
+				t.Fatalf("create consumer message is not correct: %q", string(msg.Data))
+			}
+			if ci == nil || ci.Name != "dlc-1" || ci.Stream != "foo" {
+				t.Fatalf("ConsumerInfo is not correct %+v", ci)
+			}
+		})
 
-		if _, err = js.AddConsumer("foo", &nats.ConsumerConfig{Durable: "test.durable"}); err != nats.ErrInvalidConsumerName {
-			t.Fatalf("Expected: %v; got: %v", nats.ErrInvalidConsumerName, err)
-		}
-	})
+		t.Run("with same Durable and Name set", func(t *testing.T) {
+			sub, err := nc.SubscribeSync("$JS.API.CONSUMER.CREATE.foo.dlc-2")
+			if err != nil {
+				t.Fatalf("Unexpected error: %v", err)
+			}
+			defer sub.Unsubscribe()
+			ci, err := js.AddConsumer("foo", &nats.ConsumerConfig{Durable: "dlc-2", Name: "dlc-2", AckPolicy: nats.AckExplicitPolicy})
+			if err != nil {
+				t.Fatalf("Unexpected error: %v", err)
+			}
+			msg, err := sub.NextMsg(1 * time.Second)
+			if err != nil {
+				t.Fatalf("Unexpected error: %v", err)
+			}
+			if !strings.Contains(string(msg.Data), `"durable_name":"dlc-2"`) {
+				t.Fatalf("create consumer message is not correct: %q", string(msg.Data))
+			}
+			if ci == nil || ci.Name != "dlc-2" || ci.Stream != "foo" {
+				t.Fatalf("ConsumerInfo is not correct %+v", ci)
+			}
+		})
 
-	t.Run("consumer with given name already exists", func(t *testing.T) {
-		// configs do not match
-		if _, err = js.AddConsumer("foo", &nats.ConsumerConfig{Durable: "dlc", AckPolicy: nats.AckAllPolicy}); !errors.Is(err, nats.ErrConsumerNameAlreadyInUse) {
-			t.Fatalf("Expected error: %v; got: %v", nats.ErrConsumerNameAlreadyInUse, err)
-		}
+		t.Run("with name and filter subject", func(t *testing.T) {
+			sub, err := nc.SubscribeSync("$JS.API.CONSUMER.CREATE.foo.dlc-3.foo")
+			if err != nil {
+				t.Fatalf("Unexpected error: %v", err)
+			}
+			defer sub.Unsubscribe()
+			ci, err := js.AddConsumer("foo", &nats.ConsumerConfig{
+				Durable:       "dlc-3",
+				AckPolicy:     nats.AckExplicitPolicy,
+				FilterSubject: "foo",
+			})
+			if err != nil {
+				t.Fatalf("Unexpected error: %v", err)
+			}
+			msg, err := sub.NextMsg(1 * time.Second)
+			if err != nil {
+				t.Fatalf("Unexpected error: %v", err)
+			}
+			if !strings.Contains(string(msg.Data), `"durable_name":"dlc-3"`) {
+				t.Fatalf("create consumer message is not correct: %q", string(msg.Data))
+			}
+			if ci == nil || ci.Name != "dlc-3" || ci.Stream != "foo" || ci.Config.FilterSubject != "foo" {
+				t.Fatalf("ConsumerInfo is not correct %+v", ci)
+			}
+		})
 
-		// configs are the same
-		if _, err = js.AddConsumer("foo", &nats.ConsumerConfig{Durable: "dlc", AckPolicy: nats.AckExplicitPolicy}); err != nil {
-			t.Fatalf("Expected no error; got: %v", err)
-		}
-	})
+		t.Run("ephemeral consumer without name", func(t *testing.T) {
```

Comment:
how about mentioning that this is for the legacy form in the test?
```suggestion
		t.Run("legacy ephemeral consumer without name", func(t *testing.T) {
```